str/fine-mapping/finemap_runner.py

```diff
@@ -0,0 +1,177 @@
+                            'prob': 1,
+                            'log10bf': 'NA',
+                            'mean': 'NA',
+                            'sd': 'NA',
+                            'mean_incl': 'NA',
+                            'sd_incl': 'NA',
+                        },
+                    )
+
+                    # Append the new row to the output DataFrame
+                    output_data = pd.concat([output_data, pd.DataFrame(new_row).T], ignore_index=True)
+                    output_data.to_csv(
+                        output_path(f'finemap/ofiles/{celltype}/{chrom}/{gene}.snp', 'analysis'),
+                        sep=' ',
+                        index=False,
+                    )
+                    continue
+
+                finemap_job = b.new_job(name=f'Run finemap on {celltype}:{chrom} {gene}')
+                finemap_job.image(image_path('finemap'))
+                finemap_job.cpu(job_cpu)
+                finemap_job.declare_resource_group(
+                    ofile={
+                        'data.snp': '{root}.data.snp',
+                        'data.config': '{root}.data.config',
+                        'data.log': '{root}.data.log',
+                        'data.cred': '{root}.data.cred',
+                    },
+                )
+                finemap_job.command(
+                    f"""
+                            # Create a temp_file that stores the I/O file paths for FINEMAP (as per format required by the tool)
+                            temp_file=$(mktemp)
+
+                            # Write the required format to the file
+                            echo 'z;ld;snp;config;cred;log;n_samples' > $temp_file
+                            echo "{data_in.z};{data_in.ld};{finemap_job.ofile['data.snp']};{finemap_job.ofile['data.config']};data.cred;data.log;{n_samples_total}" >> $temp_file
+                            chmod +x $temp_file
+
+                            # Run FINEMAP
+                            finemap --sss --in-files $temp_file --log --n-causal-snps {n_casual_snps_gene}
+
+                            # Concatenate all files with data.cred* into a single file
+                            cat data.cred* > {finemap_job.ofile['data.cred']}
+
+                            # Cat data.log* into a single file
+                            cat data.log* > {finemap_job.ofile['data.log']}
+
+                            """,
+                )
+                manage_concurrency_for_job(finemap_job)
+
+                output_path_vcf = output_path(f'finemap/ofiles/{celltype}/{chrom}/{gene}', 'analysis')
+                b.write_output(finemap_job.ofile['data.snp'], output_path_vcf + '.snp')
+                b.write_output(finemap_job.ofile['data.config'], output_path_vcf + '.config')
+                b.write_output(finemap_job.ofile['data.log'], output_path_vcf + '.log')
+                b.write_output(finemap_job.ofile['data.cred'], output_path_vcf + '.cred')
```

Comment:
```suggestion
                b.write_output(finemap_job.ofile, output_path_vcf)
```
partial 3/3
write_output if given a resource group will write all separate files out using their name in the declared resource group as the extension. No need to write each separately